base/strings/search.jl

```diff
@@ -212,44 +206,50 @@ julia> rsearch("aaabbb","b")
 6:6
 ```
 """
-function rsearch(s::AbstractString, c::Chars, i::Integer)
-    e = endof(s)
-    j = search(RevString(s), c, e-i+1)
-    j == 0 && return 0
-    e-j+1
+function rsearch(s::AbstractString, c::Chars, i::Integer=start(s))
+    @boundscheck checkbounds(s, i)
+    isempty(c) && return i
+    j = i
+    while j > 0
+        d = s[j]
+        d in c && return j
+        j = prevind(s, j)
+    end
+    return 0
 end
 
 function _rsearchindex(s, t, i)
     if isempty(t)
-        return 1 <= i <= nextind(s,endof(s)) ? i :
-               throw(BoundsError(s, i))
+        @boundscheck checkbounds(s, i)
+        return i
     end
-    t = RevString(t)
-    rs = RevString(s)
     l = endof(s)
-    t1, j2 = next(t,start(t))
+    t1, j2 = last(t), endof(t)
```

Comment:
`last` calls `endof`, so unless the compiler is able to elide the second call, better store the last index, and then do `t1 = t[j2]`.